OPHD/Mine.cpp

```diff
@@ -16,18 +16,18 @@ static int getOreCount(const Mine::MineVeins& veins, Mine::OreType ore, size_t d
 	int _value = 0;
 	for (size_t i = 0; i < depth; ++i)
 	{
-		_value += veins[i][ore];
+		_value += veins[i][static_cast<std::size_t>(ore)];
 	}
 	return _value;
 }
 
 
 static void setDefaultFlags(std::bitset<6>& flags)
 {
-	flags[Mine::ORE_COMMON_METALS] = true;
-	flags[Mine::ORE_COMMON_MINERALS] = true;
-	flags[Mine::ORE_RARE_METALS] = true;
-	flags[Mine::ORE_RARE_MINERALS] = true;
+	flags[static_cast<std::size_t>(Mine::OreType::ORE_COMMON_METALS)] = true;
+	flags[static_cast<std::size_t>(Mine::OreType::ORE_COMMON_MINERALS)] = true;
+	flags[static_cast<std::size_t>(Mine::OreType::ORE_RARE_METALS)] = true;
+	flags[static_cast<std::size_t>(Mine::OreType::ORE_RARE_MINERALS)] = true;
```

Comment:
The large number of `static_cast` that were introduced should perhaps have been brought up as a refactoring problem.

It would be nice to find a way to update that doesn't involve `static_cast`. That might mean a code structure change, or it might mean providing an operator on the `enum class` so that it auto converts to `std::size_t`. It should be possible to allow that to happen implicitly.

Reply:
Will also need to add conversion operators for `int` (`operator int`) as well as `std::size_t` (`operator std::size_t`). In several places the enums were used as arguments to index functions that took `int`s.

Reply:
We might also investigate if those functions should take `std::size_t` instead.